### PR TITLE
DbAdapterInterface: implemented, added quote support, factory too

### PR DESCRIPTION
**Pull request by bryanlatten**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,22 +20,21 @@ $master = [
...
 ];
 
-
-$config = new ConfigService();
-
-$config->addMaster( $master );
-$config->addReplica( $slave );
-
-$db = new ZendDbAdapter( new ConnectionService( $config ) );
+$db = new Factory::create( $config );
```
Comment (ingluisjimenez) [inline README.md]: "new" should not be required

Comment (bryanlatten) [inline README.md]: abstract away mess to build the adapter ;)

File: src/Adapters/ZendDbAdapter.php [hunk 2]
```diff
@@ -127,6 +126,16 @@ public function queryMaster( $sql, array $parameters = null, ZendResultSet $resu
   /**
    * {@inheritDoc}
    */
+  public function quote( $value ) {
```
Comment (bryanlatten) [inline src/Adapters/ZendDbAdapter.php]: missing from original spec -- needed to swap in new version of CDO...even if it is deprecated by design

File: README.md (hunk 1 above)
Comment (ingluisjimenez) [inline README.md]: but `new` can't not be used in this call, as its not an instantiation.

File: src/Interfaces/DbAdapterInterface.php [hunk 3]
```diff
@@ -0,0 +1,192 @@
+<?php
```
Comment (bryanlatten) [inline src/Interfaces/DbAdapterInterface.php]: upleveled from DbAdapterAbstract. still Zend-specific, but that's from when the pivot was made away from doctrine

File: src/Factory.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+    $config_service = $config_service ?: new ConfigService();
+
+    if ( !empty( $config['master'] ) ) {
+      $config_service->addMaster( $config['master'] );
+    }
+
+    if ( !empty( $config['replicas'] ) ) {
+
+      foreach ( $config['replicas'] as $replica ) {
```
Comment (ingluisjimenez) [inline src/Factory.php]: need to ensure that $config['replicas'] is an array

File: README.md (hunk 1 above)
Comment (bryanlatten) [inline README.md]: lol. yep

File: src/Factory.php [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+use Behance\NBD\Dbal\Services\ConfigService;
+use Behance\NBD\Dbal\Services\ConnectionService;
+use Behance\NBD\Dbal\Adapters\ZendDbAdapter;
+
+class Factory {
+
+  /**
+   * @param array $config  contains a single configuration for key 'master'
+   *                       contains an array of configurations for key 'replicas'
+   *
```
Comment (ingluisjimenez) [inline src/Factory.php]: missing @param \Behance\NBD\Dbal\Services\ConfigService $config_service

File: src/Interfaces/DbAdapterInterface.php [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+   */
+  public function bindEvent( $event_name, callable $handler );
+
+
+  /**
+   * Provided only for backwards compatibility, protects a value being entered
+   * into an SQL statement from command escaping/injection
+   *
+   * @param mixed $value
+   */
```
Comment (ingluisjimenez) [inline src/Interfaces/DbAdapterInterface.php]: missing @return

**Comment by ingluisjimenez** [comment]
:+1:

File: src/Adapters/ZendDbAdapter.php (hunk 2 above)
Comment (be-dmitry) [inline src/Adapters/ZendDbAdapter.php]: +1

**Pull request merged**